client/lib/user/user.d.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * External Dependencies
+ */
+import { EventEmitter } from 'events';
+
+export interface User extends EventEmitter {
+	initialize: () => Promise< void >;
+	clearStoreIfChanged: () => void;
+	get: () => UserData;
+	fetch: () => Promise< void > | boolean;
+	handleFetchFailure: ( error: Error ) => void;
+	handleFetchSuccess: ( userdata: UserData ) => void;
+	getLanguage: () => any | undefined;
+	getAvatarUrl: ( options: any ) => string;
```

Comment:
note that in `client/types` we have an alias for URLs…

```ts
getAvatarUrl: (options: GravatarOptions) => URL;
```

would be a decent time to put in _some_ or the _relevant_ gravatar options…

```ts
// gravatar.rs
export type GravatarOptions = Partial<{
	s: number;
	d: DefaultImage;
	f?: 'y';
	forceDefault?: 'y';
	r?: Rating;
	rating?: Rating;
}>;

type DefaultImage =
	| '404'
	| 'mp'
	| 'identicon'
	| 'monsterid'
	| 'wavatar'
	| 'retro'
	| 'robohash'
	| 'blank'
	| URL;

type Rating = 'g' | 'pg' | 'r' | 'x';
```

why go through this trouble? well…it gives people an autocomplete that shows them what options are available when they don't know in advance.

![getAvatar mov](https://user-images.githubusercontent.com/5431237/75317469-587f6680-5825-11ea-9684-12abfc5ece6f.gif)

Reply:
haha nice